### PR TITLE
Clippy fixes

### PR DESCRIPTION
**Pull request by vitvakatu**
Hello, it's my first PR for Amethyst project (and the first PR in my life too). It's pretty small and do not have any breaking changes - just few minor fixes.
First of all, I've changed one link in CONTRIBUTING.md, cause it shown an invalid code segment.
Then, I checked amethyst with [clippy](https://github.com/Manishearth/rust-clippy) and found some issues to fix. There are a lot of problems in [asset_manager](https://github.com/amethyst/amethyst/blob/develop/src/asset_manager/asset_manager.rs) and in examples, but I do not alter them, cause I'm waiting for @torkleyy 's PR.

And when I checked code with rustfmt, I found lines, which exceeded max length in lib.rs (doc line, nothing serious) and in examples. I think, it's bad, cause statement in CONTRIBUTING.md says, that code should be checked with rustfmt.

P.S. Sorry for mistakes in text, English isn't my own language.

### CODE REVIEW

**Comment by vitvakatu** [comment]
Please do not merge this PR for a while. I found some new places in code to fix.
As I said before, asset_manager.rs is awful, but I won't alter it.

**Comment by torkleyy** [comment]
@vitvakatu Please also avoid changing too much in the renderer, as it is currently reworked.

**Comment by vitvakatu** [comment]
I'm sorry, I accidentally missed one commit (between 08cadf3 and f29d830), and all changes from that commit now in f29d830. Stupid mistake, and I don't now how to fix it.
Anyway, now this PR is ready for merging.

File: src/ecs/systems/transform.rs [hunk 1]
```diff
@@ -104,7 +105,7 @@ impl System<()> for TransformSystem {
                         // If the index is none then the parent is an orphan or dead
                         if let Some(parent_index) = self.indices.get(&child.parent()) {
```
Comment (ebkalderon) [inline src/ecs/systems/transform.rs]: I think if you change `parent_index` to `ref parent_index`, then you can omit the deref below.

File: src/gfx_device/gfx_device.rs [hunk 2]
```diff
@@ -59,7 +59,7 @@ impl GfxDevice {
         // the scene.
         for (rend, entity) in (&renderables, &entities).iter() {
             let global_trans = match global_transforms.get(entity) {
-                Some(gt) => gt.clone(),
+                Some(gt) => *gt,
```
Comment (ebkalderon) [inline src/gfx_device/gfx_device.rs]: I think that if you change `Some(gt)` to `Some(ref gt)`, you can omit the deref asterisk.

File: AUTHORS.md [hunk 3]
```diff
@@ -14,3 +14,4 @@
 * Oflor
 * Robbie Cooper
 * White-Oak
+* Ilya Bogdanov
```
Comment (ebkalderon) [inline AUTHORS.md]: Perhaps you could move your name as to keep the list in alphabetical order?

Comment (vitvakatu) [inline AUTHORS.md]: Sure

File: src/gfx_device/gfx_device.rs (hunk 2 above)
Comment (vitvakatu) [inline src/gfx_device/gfx_device.rs]: I don't think it's good idea
screenshot from 2017-03-05 21 32 48

File: src/ecs/systems/transform.rs (hunk 1 above)
Comment (vitvakatu) [inline src/ecs/systems/transform.rs]: @ebkalderon I'm not sure, but when we use `ref` in this case, we have reference to reference.

File: src/gfx_device/gfx_device.rs (hunk 2 above)
Comment (ebkalderon) [inline src/gfx_device/gfx_device.rs]: Okay, thanks for checking! I don't have my computer in front of me at the moment, and I'm operating off my phone. 👍

File: src/ecs/systems/transform.rs (hunk 1 above)
Comment (ebkalderon) [inline src/ecs/systems/transform.rs]: Okay, thanks for checking! I don't have my computer in front of me at the moment, and I'm operating off my phone. 👍

**Comment by vitvakatu** [comment]
Finally, here it is.
@torkleyy, @ebkalderon thank you for reviewing.

**Comment by ebkalderon** [comment]
I think we can merge this now. Thanks for the pull request, @vitvakatu!

**Comment by ebkalderon** [comment]
I think we can merge this now. Thanks for the pull request, @vitvakatu!

**Pull request merged**